quipucords/api/credential/test_serializer.py

```diff
@@ -0,0 +1,98 @@
+
+"""Test credentials serializer."""
+import pytest
+
+from api.credential.serializer import CredentialSerializer
+from api.models import Credential
+
+
+def test_unknown_cred_type():
+    """Test if serializer is invalid when passing unknown cred type."""
+    data = {
+        "name": "cred1",
+        "cred_type": "test_cred_type",
+        "auth_token": "test_auth_token",
+    }
+    serializer = CredentialSerializer(data=data)
+    assert not serializer.is_valid(), serializer.errors
+    assert serializer.errors["cred_type"]
+
+
+def test_openshift_cred_correct_fields():
+    """Test if serializer is valid when passing mandatory fields."""
+    data = {
+        "name": "cred1",
+        "cred_type": Credential.OPENSHIFT_CRED_TYPE,
+        "auth_token": "test_auth_token",
+    }
+    serializer = CredentialSerializer(data=data)
+    assert serializer.is_valid(), serializer.errors
+    assert serializer.validated_data == data
+
+
+def test_openshift_cred_unallowed_fields():
+    """Test if serializer is invalid when passing unallowed fields."""
+    data = {
+        "name": "cred1",
+        "cred_type": Credential.OPENSHIFT_CRED_TYPE,
+        "auth_token": "test_auth_token",
+        "password": "test_password",
+        "become_password": "test_become_password",
+    }
+    serializer = CredentialSerializer(data=data)
+    assert not serializer.is_valid()
+    assert serializer.errors["password"] and serializer.errors["become_password"]
+
+
+def test_openshift_cred_empty_auth_token():
+    """Test if serializer is invalid when auth token is empty."""
+    data = {
+        "name": "cred1",
+        "cred_type": Credential.OPENSHIFT_CRED_TYPE,
+        "auth_token": "",
+    }
+    serializer = CredentialSerializer(data=data)
+    assert not serializer.is_valid(), serializer.errors
+    assert serializer.errors["auth_token"]
+
+
+def test_openshift_cred_absent_auth_token():
+    """Test if serializer is invalid when auth token is absent."""
+    data = {
+        "name": "cred1",
+        "cred_type": Credential.OPENSHIFT_CRED_TYPE,
+    }
+    serializer = CredentialSerializer(data=data)
+    assert not serializer.is_valid(), serializer.errors
+    assert serializer.errors["auth_token"]
+
+
+@pytest.mark.django_db
+def test_openshift_cred_update():
+    """Test if serializer updates fields correctly."""
+    data = {
+        "name": "cred1",
+        "cred_type": Credential.OPENSHIFT_CRED_TYPE,
+        "auth_token": "test_auth_token",
+    }
+    serializer = CredentialSerializer(data=data)
+    assert serializer.is_valid()
+
+    credential_object = serializer.save()
+    assert credential_object.name == "cred1"
+    updated_data = {
+        "name": "cred2",
+        "auth_token": "test_auth_token",
+    }
+    serializer.update(credential_object, updated_data)
+    assert serializer.is_valid()
+    assert credential_object.name == "cred2"
+    assert credential_object.cred_type == Credential.OPENSHIFT_CRED_TYPE
```

Comment:
can you please add 2 tests to the serializer?

- update w/o auth_token
- update initializing the serializer with `partial=True`

Reply:
hm, there's not much to update in openshift except for the name and auth_token. if you allow auth_token to not be passed when self.partial then how will the serializer know what has to be updated? The update method itself uses the instance to get the objects id therefore I beleive we should remove the self.partial

Reply:
How will serializer know we want to update an object and not create a new one if I only update the name per example? How can I know that the serializer I initialize with partial=True will point to the same object than before ?